src/app/item/item.component.spec.ts

```diff
@@ -1,28 +1,44 @@
 /* tslint:disable:no-unused-variable */
 import { async, ComponentFixture, TestBed } from '@angular/core/testing';
 import { By } from '@angular/platform-browser';
-import { DebugElement } from '@angular/core';
+import { DebugElement, ElementRef } from '@angular/core';
 
 import { ItemComponent } from './item.component';
+import { Item } from '../interfaces/item.interface';
 
 describe('ItemComponent', () => {
   let component: ItemComponent;
   let fixture: ComponentFixture<ItemComponent>;
-
+  let de: DebugElement;
+  let form: ElementRef<HTMLFormElement>;
   beforeEach(async(() => {
     TestBed.configureTestingModule({
-      declarations: [ ItemComponent ]
+      declarations: [ItemComponent]
     })
-    .compileComponents();
+      .compileComponents();
   }));
 
   beforeEach(() => {
     fixture = TestBed.createComponent(ItemComponent);
     component = fixture.componentInstance;
     fixture.detectChanges();
+    form = de.query(By.css('.item-form'));
   });
 
   it('should create', () => {
     expect(component).toBeTruthy();
   });
+
+  it('should update an item', () => {
+    const item: Item = {
+      itemName: 'Modified',
+      buyPrice: 1,
+      sellPrice: 2,
+      highestMarketPrice: 3
+    };
+    component.item = Object.assign({}, item);
+    spyOn(component.itemSave, 'emit');
+    form.nativeElement.submit();
+    expect(component.itemSave.emit).toHaveBeenCalledWith(item);
```

Comment:
I can't understand most what you wrote here

Reply:
basically you modify the property, use a helper method to spy on the event emitter emissions
then do the submit, the submit must trigger an emit, and should equal the item object